processing/src/test/java/io/druid/query/topn/TopNQueryRunnerTest.java

```diff
@@ -2325,4 +2330,247 @@ public void testTopNTimeExtraction()
+        .aggregators(QueryRunnerTestHelper.commonAggregators)
+        .build();
+
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    Iterator<? extends Result> result = Sequences.toList(runner.run(query, context), Lists.<Result>newArrayList())
+                                                 .iterator();
+
+    while (result.hasNext()) {
+      Result queryResult = result.next();
+      TopNResultValue value = (TopNResultValue) queryResult.getValue();
+      Assert.assertFalse(queryResult.toString(), value.getValue().isEmpty());
+    }
+  }
+
+  @Test
+  public void nullDimensionWithFilter()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.dataSource)
+        .granularity(QueryGranularity.DAY)
+        .dimension("null_column")
+        .metric(QueryRunnerTestHelper.uniqueMetric)
+        .filters(new SelectorDimFilter("null_column", null))
+        .threshold(1000)
+        .intervals(QueryRunnerTestHelper.firstToThird)
+        .aggregators(QueryRunnerTestHelper.commonAggregators)
+        .build();
+
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    Iterator<? extends Result> result = Sequences.toList(runner.run(query, context), Lists.<Result>newArrayList())
+                                                 .iterator();
+
+    while (result.hasNext()) {
+      Result queryResult = result.next();
+      TopNResultValue value = (TopNResultValue) queryResult.getValue();
+      System.out.println(value);
+      Assert.assertFalse(queryResult.toString(), value.getValue().isEmpty());
+    }
+  }
+
+  @Test
+  public void partialNullDimension()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.dataSource)
+        .granularity(QueryGranularity.DAY)
+        .dimension("partial_null_column")
+        .metric(QueryRunnerTestHelper.uniqueMetric)
+        .threshold(1000)
+        .intervals(QueryRunnerTestHelper.firstToThird)
+        .aggregators(QueryRunnerTestHelper.commonAggregators)
+        .build();
+
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    Iterator<? extends Result> result = Sequences.toList(runner.run(query, context), Lists.<Result>newArrayList())
+                                                 .iterator();
+
+    while (result.hasNext()) {
+      Result queryResult = result.next();
+      TopNResultValue value = (TopNResultValue) queryResult.getValue();
+
+      // The result can't be empty.
+      Assert.assertFalse(queryResult.toString(), value.getValue().isEmpty());
+
+      boolean containsNull = false;
+      System.out.println(value);
+      for (DimensionAndMetricValueExtractor v : value.getValue()) {
+        if (v.getStringDimensionValue("partial_null_column") == null) {
+          containsNull = true;
+        }
+
+      }
+
+      // The result always contains null dimension, and sometimes doesn't appear.
```

Comment:
What does this comment mean?  Is it saying that the code is non-deterministic and therefore we are asserting that it always contains null?